packages/apps/src/Endpoints/index.tsx

```diff
@@ -216,21 +225,29 @@ function Endpoints ({ className = '', offset, onClose }: Props): React.ReactElem
   const _onApply = useCallback(
     (): void => {
       settings.set({ ...(settings.get()), apiUrl });
-
-      window.location.assign(`${window.location.origin}${window.location.pathname}?rpc=${encodeURIComponent(apiUrl)}${window.location.hash}`);
+      window.location.assign(`${window.location.origin}${window.location.pathname}${getApiType(apiUrl).type === 'substrate-connect' as EndpointType ? '?sc=' : '?rpc='}${encodeURIComponent(apiUrl)}${window.location.hash}`);
       // window.location.reload();
-
       onClose();
     },
     [apiUrl, onClose]
   );
 
+  const isSwitchDisabled = (): boolean => {
+    if (!hasUrlChanged) { return true; }
+
+    if (getApiType(apiUrl).type === 'substrate-connect' as EndpointType) { return false; }
+
+    if (isUrlValid) { return false; }
+
+    return true;
+  };
```

Comment:
```suggestion
  function isSwitchDisabled (): boolean => {
    if (!hasUrlChanged) { 
      return true;
    } else if (getApiType(apiUrl).type === 'substrate-connect') { 
      return false
    } else if (isUrlValid) { 
      return false;
    }

    return true;
  };
```

Always use the `function` form and try to not inline returns. However, this is slightly problematic since this function is re-created each render. So really would like to avoid that at all costs.